backend/src/controllers/image-controller.ts

```diff
@@ -0,0 +1,175 @@
+import { Request, Response } from 'express'; 
+import { validationResult } from 'express-validator';
+import Image from '../models/image';
+import Item from '../models/item'; 
+import mongoose from 'mongoose';
+
+export const createImage = async (req: Request, res: Response) => {
+    // Check for validation errors 
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+        console.log("errors", errors);
+        return res.status(400).json({ errors: errors.array() });
+    } 
+
+    try {
+
+        const itemId = req.params.itemId as string
+        const imageUrl = req.body.imageUrl as string
+        const isPrimary = req.body.isPrimary as boolean
+        const order = req.body.order as number
+
+        const isIdValid = mongoose.Types.ObjectId.isValid(itemId) 
+
+        if (!isIdValid) {
+            return res.status(500).json({ message: "Invalid Credentials" }) 
+        }
+
+        const findItem = await Item.findOne({ _id: itemId }) 
+
+        const findImage = await Image.findOne({ isPrimary: true }) 
+
+        if (isPrimary && findImage) {
+            Image.findOneAndUpdate({ itemId: itemId, isPrimary: true }, { isPrimary: false }, {
+                new: true,
+                upsert: true
+            }).catch(err => { 
```

Comment:
For future reference: don't use `.catch` in `async await`

Issue is minor so I'll accept the PR